### DISCUSSION TITLE
Branch Naming Convention

### DISCUSSION THREAD

**Discussion by Arkady1356** [rbgus2002/cactusdy#236] [General]
하나의 이슈에 대해서 여러개의 브랜치가 만들어질 수 있다는 점에서

> feature/233-explaination-about-branch

와 같은 형태로 해보는 것은 어떤가요?  

<br>

예를 들어 [FE/BE]가 동시에 작업해야 하는 경우에도

같은 이슈 번호에 대해서 각각 브랜치가 생겨야 한다고 생각해요..!

(킹대차에서는 어떤식으로 하는지도 궁금합니다 ㅎㅎ)

**Comment by rbgus2002** [comment]
레포지토리를 BE/FE가 공유하며 사용하지 않고 분리해서 사용하는 것이 일반적인 것으로 알고 있어요! 같은 repo 안에서 작업할 일이 없기 때문에 BE/FE가 각각 다른 컨벤션을 가져가는 경우도 종종 본 거 같습니다.

사실 github issue와 git branch는 1:1로 만들어 개발하는 것이 취지였어요..! (물론 귀찮아서 룰을 지키지 않은 적이 꽤 있긴한데 ㅎㅎ;)
그래서 FE/BE가 동시에 작업해야하는 경우를 만들지말고 이슈를 각각 따로 발행해서 작업하는 건 어떠세요?
어차피 삭제될 branch에 문자를 추가해서 네이밍하는 게 의미없는 일이라고 생각이 들어서요 ! 

그리고 각각 따로 이슈를 따야 나중에 이슈 추적하기 편할 거라고 생각듭니당
가령 제가 #232 에서 작업한 내용을 가지고 merge 한 뒤에 #233 에서 이슈를 따로 따서 assign 해드렸는데요, 이런 식으로 가야지 이슈가 많아졌을 때 관리하기에 좋을 거 같아요!

의견 주시면 감사하겠습니다 ~

**Reply by Arkady1356** [reply]
흐으음 그렇군요 
개인적인 생각으로는 #232와 #233으로 나누었을때 차후 History F/Up에 불편함이 있을 것 같긴합니다..!

Issue가 최초에부터 FE/BE로 나누어져서 나오는게 아니라
어떤 현상이 발생하고, 그 현상을 해결하다 보니 FE 혹은 BE에 수정이 필요해진다고 생각했습니다..!
앞으로 FE와 BE가 동시에 수정될 일 도 많이 있을 것이라 생각됩니다

근데 뭐 우리가 Histroy F/Up이 필요한 일이 많을 것 같지도 않고 ...
말씀하신 것 처럼 각각 이슈 발행하는 편이 좋을 것 같네요 ㅎㅎㅎ

**Comment by Arkady1356** [comment]
선생님 개발 브랜치 명이 숫자로만 있으니까 불편해요

**Reply by rbgus2002** [reply]
깃 브랜치 이름만 보고 브랜치의 용도를 파악하려다 보니 불편함이 생기신 것 같네여
본문에서 언급한 것처럼, 해당 브랜치가 수행하는 작업을 요약해서 포함하는 것이 더 좋다는 말씀이시져??
저는 이슈 번호 보고 브랜치 구분하기 때문에 크게 상관 없으니 본문처럼 컨벤션 수정할까요 ?.?

**Reply by rbgus2002** [reply]
저는 본문처럼 해도 좋습니다 ㅎㅎ

**Reply by Arkady1356** [reply]
그 회사는 다들 천재인건가..? 어케 번호 만으로 피처를 구분하시는거죠..?! ;ㅅ; 넘사벽이다

**Reply by rbgus2002** [reply]
ㄷㄷ.. 당연히 깃헙 가서 발행한 이슈 봐야져 ㅋㅌㅋㅌㅋ
